staging/src/k8s.io/kubectl/pkg/cmd/apply/applyset.go

```diff
@@ -141,10 +143,24 @@ func NewApplySet(parent *ApplySetParentRef, tooling ApplySetTooling, mapper meta
 	}
 }
 
+const applySetIDPartDelimiter = "."
+
 // ID is the label value that we are using to identify this applyset.
+// Format: base64(sha256(<name>.<namespace>.<kind>.<group>)), using the URL safe encoding of RFC4648.
 func (a ApplySet) ID() string {
-	// TODO: base64(sha256(gknn))
-	return "placeholder-todo"
+	var unencoded string
+	if a.parentRef.IsNamespaced() {
+		unencoded = strings.Join([]string{a.parentRef.Name, a.parentRef.Namespace, a.parentRef.GroupVersionKind.Kind, a.parentRef.GroupVersionKind.Group}, applySetIDPartDelimiter)
+	} else {
+		unencoded = strings.Join([]string{a.parentRef.Name, a.parentRef.GroupVersionKind.Kind, a.parentRef.GroupVersionKind.Group}, applySetIDPartDelimiter)
+	}
+	w := bytes.Buffer{}
+	encoder := base64.NewEncoder(base64.URLEncoding, &w)
+	_, err := encoder.Write([]byte(unencoded))
+	if err != nil {
+		klog.Fatalf("failed to encode parent ID %s: %w", unencoded, err)
```

Comment:
Is this unlikely enough that the panic is ok? Otherwise we'll need error handling everywhere we use the ID.

Reply:
Can we use `EncodeToString`?  Then you avoid the issue (or rather let the base64 library make the assumption!)  Your assumption is valid here though, I believe.

It looks like the sha256 is also missing BTW, and there I would similarly use `Sum256` to avoid error handling